packages/deepgram/src/deepgram-transcription-model.ts

```diff
@@ -0,0 +1,245 @@
+          path: '/v1/listen',
+          modelId: this.modelId,
+        }) +
+        '?' +
+        queryParams.toString(),
+      headers: {
+        ...combineHeaders(this.config.headers(), options.headers),
+        'Content-Type': options.mediaType,
+      },
+      body: {
+        content: options.audio,
+        values: options.audio,
+      },
+      failedResponseHandler: deepgramFailedResponseHandler,
+      successfulResponseHandler: createJsonResponseHandler(
+        deepgramTranscriptionResponseSchema,
+      ),
+      abortSignal: options.abortSignal,
+      fetch: this.config.fetch,
+    });
+
+    return {
+      text:
+        response.results?.channels.at(0)?.alternatives.at(0)?.transcript ?? '',
+      segments:
+        response.results?.channels[0].alternatives[0].words?.map(word => ({
+          text: word.word,
+          startSecond: word.start,
+          endSecond: word.end,
+        })) ?? [],
+      language: undefined,
+      durationInSeconds: response.metadata?.duration ?? undefined,
+      warnings,
+      response: {
+        timestamp: currentDate,
+        modelId: this.modelId,
+        headers: responseHeaders,
+        body: rawResponse,
+      },
+    };
+  }
+}
+
+const deepgramTranscriptionResponseSchema = z.object({
+  metadata: z
+    .object({
+      transaction_key: z.string(),
+      request_id: z.string(),
+      sha256: z.string(),
+      created: z.string(),
+      duration: z.number(),
+      channels: z.number(),
+      models: z.array(z.string()),
+      model_info: z.record(
+        z.object({
+          name: z.string(),
+          version: z.string(),
+          arch: z.string(),
+        }),
+      ),
+    })
+    .nullish(),
+  results: z
+    .object({
+      channels: z.array(
+        z.object({
+          alternatives: z.array(
+            z.object({
+              transcript: z.string(),
+              confidence: z.number(),
+              words: z.array(
+                z.object({
+                  word: z.string(),
+                  start: z.number(),
+                  end: z.number(),
+                  confidence: z.number(),
+                  punctuated_word: z.string().nullish(),
+                }),
+              ),
+              paragraphs: z
+                .object({
+                  transcript: z.string(),
+                  paragraphs: z.array(
+                    z.object({
+                      sentences: z.array(
+                        z.object({
+                          text: z.string(),
+                          start: z.number(),
+                          end: z.number(),
+                        }),
+                      ),
+                      num_words: z.number(),
+                      start: z.number(),
+                      end: z.number(),
+                    }),
+                  ),
+                })
+                .nullish(),
+            }),
+          ),
+        }),
+      ),
+    })
+    .nullish(),
+});
```

Comment:
are all these properties used? if not, removing unused ones has 3 benefits:
- smaller bundle size
- faster validation
- increased resilience against API changes